### PR TITLE
Migrate Actions and Reactions to Proto

### PR DESCRIPTION
**Pull request by timothyfroehlich**

### CODE REVIEW

File: crates/figma_import/src/component_context.rs [hunk 1]
```diff
@@ -53,19 +53,16 @@ impl ComponentContext {
...
-            // anything with those.
-            let destination_node_id = match &reaction.action {
-                Action::Node { destination_id: Some(id), .. } => id,
-                _ => continue,
-            };
-
-            // If we've already converted this node then don't add it again.
-            if self.converted_nodes.contains(destination_node_id) {
-                continue;
+            // // anything with those.
```
Comment (yiqunw700) [inline crates/figma_import/src/component_context.rs:56]:
```suggestion
            // anything with those.
```

**Comment by yiqunw700** [comment]
I tried the interaction test. It seems the overlays are broken. I can not see any overlays. Could you try on your end, the `Interaction` -> `Overlay tests`?

**Comment by timothyfroehlich** [comment]
Oi, yeah, they're not working. Let me try to add some tests that'll catch that, and then I'll fix them.

File: designcompose/src/main/java/com/android/designcompose/DesignView.kt [hunk 2]
```diff
@@ -309,19 +315,25 @@ internal fun DesignView(
...
+            reaction.trigger.type?.run {
+                when (this) {
+                    is TriggerType.Click -> onClickReactions.add(reaction)
+                    is TriggerType.Press -> onPressReactions.add(reaction)
+                    is TriggerType.Drag -> onDragReactions.add(reaction)
+                    is TriggerType.AfterTimeout -> {
+                        if (value.timeout < currentTimeout) {
+                            onTimeout = reaction
+                            currentTimeout = value.timeout
+                        } else {
```
Comment (yiqunw700) [inline designcompose/src/main/java/com/android/designcompose/DesignView.kt:330]: We don't really need this `else {{}}` right? Remove?

Comment (timothyfroehlich) [reply designcompose/src/main/java/com/android/designcompose/DesignView.kt:329]: The extra {} isn't needed, but the else is needed when used in a `when`, apparently.

**Pull request merged**